### ISSUE TITLE
Support for Internationalized (i18n) automatic locale detection

### ISSUE THREAD

**Issue by ArianHamdi** [aws-amplify/amplify-hosting#3320]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting questions](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d2kp69mengqx1w

### AWS Region

us-east-1

### Amplify Hosting feature

Deployments, SSR

### Question

Our next.js 13 application is being migrated to the new amplify hosting compute platform.
We need to configure cloud front for two reasons:
1: to redirect "domain.com/restApi/*" to the API gateway so that they won't be resolved in Amplify Hosting.
2: to get the 'Accept-Language' header in middleware to display the website in the user's preferred language. 

We previously added path behavior to cloudfront to point "/restApi/*" to an API gateway and let cloudfront distribution allow "Accept-Language" header.

As of now, amplify has full control over cloudfront and we are unsure how to resolve this issues.

Thanks in advance

**Comment by hloriana** [comment]
Hi @ArianHamdi 👋🏽, thanks for raising this question and explaining your use case for these configurations. We do not support modifications to the CloudFront distribution at this time since it is managed by Amplify with the new compute platform. However, we are tracking this as a feature request.

**Comment by ArianHamdi** [comment]
Hi @hloriana.
[Amplify sipport for Next.js SSR](https://docs.aws.amazon.com/amplify/latest/userguide/ssr-Amplify-support.html) points to automatic locale detection, but I was not able to implement it.

`next.config.js` has a localeDetection option that automatically redirects users to their preferred language, but the feature does not work on Amplify.

It seems to rely on Accept-Language header, which is blocked by Amplify (It's unclear to me).

On introduce amplify for next.js, I'm not sure why automatic locale detection is added if this header is blocked.

Please let me know if this comment needs a new issue.

**Comment by vafPER** [comment]
@hloriana Do you have any updates about the headers? Http request headers like "Accept-Language" or "Referer" seems not to be passed through Amplify Host. We are stuck in development after migrating our front end to Amplify.

**Comment by vafPER** [comment]
It's hilarious and at the same time sad that we got to know today that what was written in AWS Amplify doc was wrong and Automatic locale detection for Next Js is not supported yet. At least we managed to make them correct their official page :

https://docs.aws.amazon.com/amplify/latest/userguide/ssr-Amplify-support.html

so no other team falls into the trap of spending days on finding the cause of an error which is written wrongly as a supported feature.

**Comment by burkayanduv** [comment]
Are there any updates on this? 

Everything was seamless for hosting Next.js app with Amplify until we have learned the accept-language headers were stripped off from requests by CloudFront and there is no way to configure this behavior. We will stick with Vercel until this one is fixed. Hope this issue is resolved in a timely manner.

**Comment by gstrobl** [comment]
Any updates for this feature?

**Comment by mdillgen** [comment]
+1 This feature request is very important and the lack of language detection (in my case handled by middleware inside of Next.js 14) is what is currently holding me back from hosting on amplify.

**Comment by snaps-jacob** [comment]
try 'cloudfront-viewer-country' field
but it is not perfect.

**Comment by mauerbac** [comment]
This feature has just been supported with our latest launch! Closing.

Check out the details here: https://aws.amazon.com/blogs/mobile/cdn-caching-improvements-for-better-app-performance-with-aws-amplify-hosting/
